Python/Product/Analysis/Parsing/Tokenizer.cs

```diff
@@ -1307,28 +1321,37 @@ public Tokenizer(PythonLanguageVersion version, ErrorSink errorSink = null, Toke
                     case 'J':
                         MarkTokenEnd();
 
-                        // TODO: parse in place
+                        tokenStr = GetTokenString();
                         if (Verbatim) {
-                            string tokenStr = GetTokenString();
-                            return new VerbatimConstantValueToken(LiteralParser.ParseImaginary(tokenStr), tokenStr);
+                            return new VerbatimConstantValueToken(ParseComplex(tokenStr), tokenStr);
                         }
-                        return new ConstantValueToken(LiteralParser.ParseImaginary(GetTokenString()));
+                        return new ConstantValueToken(ParseComplex(tokenStr));
+
+                    case '_':
+                        if (LanguageVersion < PythonLanguageVersion.V36) {
+                            goto default;
+                        }
+                        break;
 
                     default:
                         BufferBack();
                         MarkTokenEnd();
 
-                        // TODO: parse in place
+                        tokenStr = GetTokenString();
                         if (Verbatim) {
-                            string tokenStr = GetTokenString();
                             return new VerbatimConstantValueToken(ParseFloat(tokenStr), tokenStr);
                         }
-                        return new ConstantValueToken(ParseFloat(GetTokenString()));
+                        return new ConstantValueToken(ParseFloat(tokenStr));
                 }
             }
         }
 
         private Token ReadExponent(bool leftIsFloat = false) {
+            BufferBack();
+            MarkTokenEnd();
+            var beforeEStr = GetTokenString();
```

Comment:
unused variable?

Reply:
Yeah, was a nice but ultimately insufficient idea to validate better.